src/react/useBaseQuery.ts

```diff
@@ -91,5 +91,9 @@ export function useBaseQuery<TQueryFnData, TError, TData, TQueryData>(
     }
   }
 
-  return currentResult
+  observer.resetTrackedProps()
```

Comment:
While I do like the dynamic tracking, would it not "forget" tracked properties in `useEffect` for example?
```js
const result = useQuery()

useEffect(() => {
  if (result.error) {
    // do something
  }
}, [result])

return <>{result.data}</>
```

Reply:
You're right, this  is similar to the callback case I was showing above. You'd need to change the dependencies to include `error`:

```diff
useEffect(() => {
  if (result.error) {
    // do something
  }
-}, [result])
+}, [result.error])
```

this is also what the linter suggests when you add an empty array at the let the auto-fix add it.

Do you think it's worth _not_ resetting because of this?

Reply:
Think it's at least a good reason to not enable it by default as this could break applications. Apart from that: not resetting would probably be the safest option because if we would reset, bugs like this could easily sneak in. Maybe the behavior should be configurable with proper documentation so people know the implications?

Reply:
Yes, I agree on not enabling by default. The callback case is something we can't solve with this I believe. I'm happy to change it to not resetting for now, it's the "safer" case because it means an unnecessary re-render at worst rather than a bug :)

Reply:
auto resetting removed here: 14940909